Sources/Purchasing/StoreKit2/StoreKit2TransactionFetcher.swift

```diff
@@ -90,15 +115,147 @@ extension StoreKit.VerificationResult where SignedType == StoreKit.Transaction {
     var verifiedTransaction: StoreKit.Transaction? {
         switch self {
         case let .verified(transaction): return transaction
-        case .unverified: return nil
+        case let .unverified(transaction, error):
+            Logger.warn(Strings.storeKit.sk2_unverified_transaction(identifier: String(transaction.id), error))
+            return nil
         }
     }
 
-    fileprivate var verifiedStoreTransaction: StoreTransaction? {
+    var verifiedStoreTransaction: StoreTransaction? {
         switch self {
         case let .verified(transaction): return StoreTransaction(sk2Transaction: transaction,
                                                                  jwsRepresentation: self.jwsRepresentation)
-        case .unverified: return nil
+        case let .unverified(transaction, error):
+            Logger.warn(Strings.storeKit.sk2_unverified_transaction(identifier: String(transaction.id), error))
+            return nil
+        }
+    }
+
+}
+
+// MARK: - Private
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, watchOS 9.0, *)
+extension StoreKit.VerificationResult where SignedType == StoreKit.AppTransaction {
+
+    var verifiedAppTransaction: SK2AppTransaction? {
+        switch self {
+        case let .verified(transaction): return .init(appTransaction: transaction)
+        case let .unverified(transaction, error):
+            Logger.warn(
+                Strings.storeKit.sk2_unverified_transaction(identifier: transaction.bundleID, error)
+            )
+            return nil
+        }
+    }
+
+}
+
+extension StoreKit2TransactionFetcher {
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    private var verifiedTransactions: [StoreTransaction] {
+        get async {
+            return await StoreKit.Transaction.all
+                .compactMap { $0.verifiedStoreTransaction }
+                .extractValues()
+        }
+    }
+
+    @available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *)
+    func verifiedTransactions(containing transaction: StoreTransactionType) async -> [StoreTransaction] {
```

Comment:
Let's add a docstring explaining what this does, I don't think it's obvious reading the signature.